redis/connection.py

```diff
@@ -548,9 +550,11 @@ def __init__(
         self.host = host
         self.port = int(port)
         self.db = db
-        self.username = username
         self.client_name = client_name
-        self.password = password
+        self.credential_provider = credential_provider
+        if (username or password) and self.credential_provider is None:
+            # username and password backward compatibility
+            self.credential_provider = StaticCredentialProvider(username, password)
```

Comment:
I think it'd be wise to raise an error if both username+password _and_ a provider are passed in, seeing as they smell mutually exclusive.

Reply:
added